Source/modules/webcl/WebCLOpenCL.cpp

```diff
@@ -0,0 +1,275 @@
+
+cl_int (CL_API_CALL *web_clEnqueueCopyBufferRect)(cl_command_queue command_queue, cl_mem src_buffer, cl_mem dst_buffer, const size_t* src_origin, const size_t* dst_origin, const size_t* region, size_t src_row_pitch, size_t src_slice_pitch, size_t dst_row_pitch, size_t dst_slice_pitch, cl_uint num_events_in_wait_list, const cl_event* event_wait_list, cl_event* event);
+
+cl_int (CL_API_CALL *web_clEnqueueCopyImage)(cl_command_queue command_queue, cl_mem src_image, cl_mem dst_image, const size_t* src_origin, const size_t* dst_origin, const size_t* region, cl_uint num_events_in_wait_list, const cl_event* event_wait_list, cl_event* event);
+
+cl_int (CL_API_CALL *web_clEnqueueCopyImageToBuffer)(cl_command_queue command_queue, cl_mem src_image, cl_mem dst_buffer, const size_t* src_origin, const size_t* region, size_t dst_offset, cl_uint num_events_in_wait_list, const cl_event* event_wait_list, cl_event* event);
+
+cl_int (CL_API_CALL *web_clEnqueueCopyBufferToImage)(cl_command_queue command_queue, cl_mem src_buffer, cl_mem dst_image, size_t src_offset, const size_t* dst_origin, const size_t* region, cl_uint num_events_in_wait_list, const cl_event* event_wait_list, cl_event* event);
+
+/* OpenCL Extention */
+cl_int (CL_API_CALL *web_clEnqueueAcquireGLObjects)(cl_command_queue command_queue, cl_uint num_objects, const cl_mem* mem_objects, cl_uint num_events_in_wait_list, const cl_event* event_wait_list, cl_event* event);
+
+cl_int (CL_API_CALL *web_clEnqueueReleaseGLObjects)(cl_command_queue command_queue, cl_uint num_objects, const cl_mem* mem_objects, cl_uint num_events_in_wait_list, const cl_event* event_wait_list, cl_event* event);
+
+cl_mem (CL_API_CALL *web_clCreateFromGLBuffer)(cl_context context, cl_mem_flags flags, GLuint bufobj, cl_int* errcode_ret);
+
+cl_mem (CL_API_CALL *web_clCreateFromGLRenderbuffer)(cl_context context, cl_mem_flags flags, GLuint renderbuffer, cl_int* errcode_ret);
+
+cl_mem (CL_API_CALL *web_clCreateFromGLTexture)(cl_context context, cl_mem_flags flags, GLenum texture_target, GLint miplevel, GLuint texture, cl_int* errcode_ret);
+
+cl_int (CL_API_CALL *web_clGetGLTextureInfo)(cl_mem, cl_gl_texture_info, size_t, void *, size_t *);
+
+#if defined(WTF_OS_LINUX) || OS(ANDROID)
+#define MAP_FUNC(fn)  { *(void**)(&fn) = dlsym(handle, #fn); if(!fn) return false; }
+
+static const char* DEFAULT_SO[] = LIBS;
+static const int DEFAULT_SO_LEN = SO_LEN;
+static void* handle = nullptr;
+static bool getCLHandle(const char** libs, int length)
+{
+    for (int i = 0; i < length; ++i) {
+        handle = dlopen(libs[i], RTLD_LAZY);
+        if (handle)
+            return true;
+    }
+
+    /* FAILURE: COULD NOT OPEN .SO */
+    return false;
+}
+#endif // defined(WTF_OS_LINUX) || OS(ANDROID)
+
+bool init(const char** libs, int length)
+{
+    const char** mLibs = (libs == 0 ? DEFAULT_SO : libs);
+    int mLength = (libs == 0 ? DEFAULT_SO_LEN: length);
+
```

Comment:
Because we have define "#define MAP_FUNC(fn)  { *(void**)(&fn) = dlsym(handle, #fn); if(!fn) return false; }"

So the library is't compatible with pre-defined apis in code, it would failed to load library.
I think it's more strict than version check.